File: Array/wuchao/stack/6-19删除字符串中所有相邻的字符/RemoveDuplicatesSolution.java
package com.slow.lib.practice.stack;

import java.util.Stack;

/**
 * @Author wuchao
 * @Date 2020/6/19-9:20 AM
 * @description
 * @email 329187218@qq.com
 * @see
 */
public class RemoveDuplicatesSolution {
    public String removeDuplicates(String S) {
        int stringLength = S.length();
        Stack<Character> stack = new Stack<>();
        int i = 0;
        while(i<stringLength){
            char item = S.charAt(i);
            if(stack.isEmpty()){
                stack.push(item);
            } else {
                if(stack.peek()==item){
                    stack.pop();
                }else{
                    stack.push(item);
                }
            }
            i++;
        }
        StringBuilder result = new StringBuilder();
        for(Character item:stack){
            result.append(item);
        }
        return result.toString();
    }
}

File: Array/wuchao/stack/6-19删除字符串中所有相邻的字符/RemoveDuplicatesSolution.kt
package com.slow.lib.practice.stack

import java.util.*

/**
 * @Author wuchao
 * @Date 2020/6/19-9:48 AM
 * @see
 * @description
 * @email 329187218@qq.com
 */
class RemoveDuplicatesSolution {
    fun removeDuplicates(S: String): String {
        val stack = Stack<Char>()
        for(i in 0..S.length-1){
            if(stack.isEmpty()){
                stack.push(S[i])
            } else {
                if(stack.peek()==S[i]){
                    stack.pop()
                } else {
                    stack.push(S[i])
                }
            }
        }
        val stringBuilder = StringBuilder()
        for(item in stack){
            stringBuilder.append(item)
        }
        return stringBuilder.toString()
    }
}